sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -431,11 +437,11 @@ private static Expression value(
     private static Expression value(Expression value, Schema.FieldType type) {
       if (type.getTypeName().isLogicalType()) {
         String logicalId = type.getLogicalType().getIdentifier();
-        if (TimeType.IDENTIFIER.equals(logicalId)) {
+        if (SqlTypes.TIME.getIdentifier().equals(logicalId)) {
           return nullOr(
-              value, Expressions.convert_(Expressions.call(value, "getMillis"), int.class));
+              value, Expressions.divide(value, Expressions.constant(NANOS_PER_MILLISECOND)));
         } else if (SqlTypes.DATE.getIdentifier().equals(logicalId)) {
-          value = nullOr(value, value);
+          return nullOr(value, value);
```

Comment:
This is going to generate the code `value == null ? null : value`, which does nothing. drop it all together?

Reply:
`nullOr()` also does boxing for `value` if it is not null. Is it necessary in this case? I tried to remove the call to `nullOr()` and it works fine as well.

Reply:
Yeah. It works if we just return value here.